clang/include/clang/Lex/HeaderSearch.h

```diff
@@ -124,7 +124,7 @@ struct HeaderFileInfo {
   /// This ID number will be non-zero when there is a controlling
   /// macro whose IdentifierInfo may not yet have been loaded from
   /// external storage.
-  unsigned ControllingMacroID = 0;
+  uint64_t ControllingMacroID = 0;
```

Comment:
This increases the size of `HeaderFileInfo` from 32 to 40 bytes. Can we squash this member with `ControllingMacro` into some kind of tagged pointer & `uint64_t` union to save space?

Reply:
Done by adding `LazyIdentifierInfoPtr` to `ExternalPreprocessorSource.h`.